src/aruco_pose/src/aruco_pose.cpp

```diff
@@ -0,0 +1,218 @@
+  // // These matrices are already available in the topic '/rgb/camera_info',
+  // // But it doesn't make sense to have two subscribers running to get static parameters.
+  // // Plus this avoids a situation of a different topic name being used by another vendor
+
+  // Declare parameters
+  for (const auto & param : double_params_) {
+    this->declare_parameter<double>(param, 0.0);
+  }
+  for (const auto & param : string_params_) {
+    this->declare_parameter<std::string>(param, "");
+  }
+
+  this->declare_parameter<int>("number_of_observations", 10);
+
+  cameraMatrix_ =
+    (cv::Mat_<double>(3, 3) << this->get_parameter("intrinsics.fx").as_double(), 0.0,
+    this->get_parameter("intrinsics.cx").as_double(), 0.0,
+    this->get_parameter("intrinsics.fy").as_double(),
+    this->get_parameter("intrinsics.cy").as_double(), 0, 0, 1.0);
+
+  distCoeffs_ =
+    (cv::Mat_<double>(8, 1) << this->get_parameter("dist_coeffs.k1").as_double(),
+    this->get_parameter("dist_coeffs.k2").as_double(),
+    this->get_parameter("dist_coeffs.p1").as_double(),
+    this->get_parameter("dist_coeffs.p2").as_double(),
+    this->get_parameter("dist_coeffs.k3").as_double(),
+    this->get_parameter("dist_coeffs.k4").as_double(),
+    this->get_parameter("dist_coeffs.k5").as_double(),
+    this->get_parameter("dist_coeffs.k6").as_double());
+
+  physical_marker_size_ = this->get_parameter("physical_marker_size").as_double();
+
+  double alpha = this->get_parameter("cam_rotation.alpha").as_double() / 180 * M_PI;
+  double beta = this->get_parameter("cam_rotation.beta").as_double() / 180 * M_PI;
+  double gamma = this->get_parameter("cam_rotation.gamma").as_double() / 180 * M_PI;
+
+  // Add the camera to tf server
+  camera_quaternion_.setRPY(alpha, beta, gamma);
+
+  // Define the transform
+  geometry_msgs::msg::TransformStamped transformStamped;
+  transformStamped.header.stamp = this->now();
+  transformStamped.header.frame_id = "world";
+  transformStamped.child_frame_id = this->get_parameter("camera_tf_frame").as_string();
+  transformStamped.transform.translation.x = this->get_parameter("cam_translation.x").as_double();
+  transformStamped.transform.translation.y = this->get_parameter("cam_translation.y").as_double();
+  transformStamped.transform.translation.z = this->get_parameter("cam_translation.z").as_double();
+  transformStamped.transform.rotation.x = this->camera_quaternion_.x();
+  transformStamped.transform.rotation.y = this->camera_quaternion_.y();
+  transformStamped.transform.rotation.z = this->camera_quaternion_.z();
+  transformStamped.transform.rotation.w = this->camera_quaternion_.w();
+
+  static_broadcaster_.sendTransform(transformStamped);
+
+  camera_subscription_ = this->create_subscription<sensor_msgs::msg::Image>(
+    "/rgb/image_raw", 5,
+    std::bind(&ArucoPose::image_raw_callback, this, std::placeholders::_1));
```

Comment:
`/rgb/image_raw` should be a default string that is parameterized, not a compiled string. 

The polling frequency can also be a param

Reply:
The second parameter in this->create_subscription is not the polling frequency. It is the message queue size. Polling rate is defined by the message publisher. In Azure ROS package, where the topic 'rgb/image_raw' is published, message publish rate is defined by the camera's fps rate. Default fps is 5.

Reply:
parameterized and readme updated:
https://github.com/maffettone/erobs/blob/5c32dc6fe6c0a002d6ecb2d3e5b09daddd49ae45/src/aruco_pose/config/fiducial_marker_param.yaml#L22

https://github.com/maffettone/erobs/blob/5c32dc6fe6c0a002d6ecb2d3e5b09daddd49ae45/src/aruco_pose/README.md?plain=1#L20-L24